### ISSUE TITLE
Incomplete UTF-8 support in Android App (Devanagari Script)

### ISSUE THREAD

**Issue by rnayabed** [gluonhq/gluonfx-maven-plugin#382]
Internationalisation with resource bundle works fine for Spanish, but languages like Hindi/Marathi does not work.

Non native-image (JVM):

![image](https://user-images.githubusercontent.com/25760501/139389356-884932e8-161b-47c6-9ad8-14f5fad05ba3.png)
![image](https://user-images.githubusercontent.com/25760501/139389432-23690b0c-b355-4e23-8557-7e035c3924bd.png)

GluonFX Android build: 
![photo_2021-10-29_12-26-53](https://user-images.githubusercontent.com/25760501/139389743-89f94440-c41b-4589-b67c-52b72a1dcfb0.jpg)
![photo_2021-10-29_12-26-49](https://user-images.githubusercontent.com/25760501/139389736-3a1c4a3a-dcf1-4421-aa6e-a60fae12118e.jpg)

**Comment by jperedadnr** [comment]
I've just tried Stream-Pi on Mac and Linux with JDK 11 (or 13), I have a crash simply running `mvn javafx:run`:
```
Calling main(String[]) method
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x000000010bcd87c3, pid=8907, tid=259
#
# JRE version: OpenJDK Runtime Environment (11.0.2+7) (build 11.0.2+7)
# Java VM: OpenJDK 64-Bit Server VM (11.0.2+7, mixed mode, tiered, compressed oops, g1 gc, bsd-amd64)
# Problematic frame:
# V  [libjvm.dylib+0x4d87c3]  klassItable::initialize_itable_for_interface(int, Klass*, bool, Thread*)+0x2b
```
but it works with JDK 16/17.

When running native-image, it works on Mac and Linux with JDK 11. I can select all the different languages. On Linux it works perfectly fine, on Mac I have some issues with the last two (wrong encoding, which I can reproduce as well with `javafx:run` and JDK17).

Anyway, running on Android, I get the same result as your screenshots, and logs show:
```
D/GraalCompiled(29378): Only simple text supported.
```
That comes from this [JavaFX code](https://github.com/openjdk/jfx/blob/master/modules/javafx.graphics/src/main/java/com/sun/javafx/font/freetype/HBGlyphLayout.java#L37). 

Java supports Harfbuzz on Mac/Linux but not on iOS/Android, where only FreeType is supported.

**Comment by rnayabed** [comment]
> I have a crash simply running `mvn javafx:run`

This is weird. I go around testing and running it via the same command all the time and it works just fine.

I am using jdk 17
```
openjdk version "17" 2021-09-14
OpenJDK Runtime Environment Temurin-17+35 (build 17+35)
OpenJDK 64-Bit Server VM Temurin-17+35 (build 17+35, mixed mode, sharing)
```

**Comment by jperedadnr** [comment]
Yes, 17 works. Probably this issue: https://bugs.openjdk.java.net/browse/JDK-8226798, solved since JDK 14, and backported to 11.0.5.

GraalVM with JDK11 already has 11.0.11 so it includes the fix.

**Comment by rnayabed** [comment]
>  I have some issues with the last two (wrong encoding, which I can reproduce as well with javafx:run and JDK17).

All the resource bundle files are stored as UTF-8. Can you please elaborate on this if possible?

> Java supports Harfbuzz on Mac/Linux but not on iOS/Android, where only FreeType is supported.

So what can be possibly done at the moment? 🤔

**Comment by jperedadnr** [comment]
On Mac with JDK17 and JavaFX 18-ea+5:

<img width="812" alt="image" src="https://user-images.githubusercontent.com/2043230/139483510-8cbdd072-71c5-483c-bdb4-b31c1c71f407.png">


<img width="812" alt="image" src="https://user-images.githubusercontent.com/2043230/139483471-bce094ad-e398-4d22-afaf-22fff8e0dbf9.png">

As for Android, I'd start by doing a small test: just a simple JavaFX app with a combobox that allows you to select any possible font from the collection `Font.getFontNames()`, set locale accordingly, and pick one by one the different fonts from the combo and see if you can find one that actually renders some text?

**Comment by rnayabed** [comment]
@jperedadnr 

Actually if you look at the title bar of your screenshot, the desired titlebar is achieved ( Stream-Pi क्लाइंट)

But the main content of the javafx window is what is broken. So this looks more like javafx unable to select the appropriate font rather than bad encoding ...

**Comment by jperedadnr** [comment]
The title bar of the window is actually managed by the system, so that explains why it works. 

Recently I fixed probably this very same issue with this PR: https://github.com/openjdk/jfx/pull/399

Not sure if you can verify that those two languages are supported properly by Java/JavaFX on macOS?

**Comment by rnayabed** [comment]
> The title bar of the window is actually managed by the system, so that explains why it works.
> 
> Recently I fixed probably this very same issue with this PR: [openjdk/jfx#399](https://github.com/openjdk/jfx/pull/399)
> 
> Not sure if you can verify that those two languages are supported properly by Java/JavaFX on macOS?

The thing is that it works perfectly fine with normal JVM + linux/windows. and is just looking like something specific to MacOS. The myanmar script and hindi (devanagari) script is not the same. 

I dont have a MacOS system with me. I will ask some more people in the streampi community with macos to test it out

**Comment by jperedadnr** [comment]
I've run a simple test case with strings from your properties files:
```
    @Override
    public void start(final Stage primaryStage) {
        final Label btn = new Label("सर्वर होस्टनाम/आईपी");
//        final Label btn = new Label("सर्वर पोर्ट का मूल्य पूर्णांक होना चाहिए।");
        StackPane root = new StackPane(btn);
        primaryStage.setScene(new Scene(root, 300, 250));
        primaryStage.show();
    }

    public static void main(String[] args) {
        System.setProperty("prism.debugfonts", "true");
        launch(args);
    }
```
And that works fine for me on macOS.

The output log is interesting:
```
[MacFontFinder] Name=System Font Bold
        Family=.AppleSystemUIFont
        File=/System/Library/Fonts/SFNS.ttf
        Fallback font= Kohinoor Devanagari Regular slot=6
 ...
```
So I haven't set any font for that label, so the System one is applied, and the font finder finds the correct font.

Back to Stream-Pi client, I see that you use  `Roboto` font. So I've removed that, and now the System font gets applied and the i18n panel works perfectly fine.  

<img width="812" alt="image" src="https://user-images.githubusercontent.com/2043230/139561651-29345e46-f6af-46b7-93f6-8f3b82d9b851.png">

**Comment by rnayabed** [comment]
I had actually removed "Roboto" as the default font completely from Stream-Pi. However, I forgot to remove the CSS part where `-fx-font-family: Roboto` was still written. 

This is probably why the Server worked just fine, because I removed all roboto entirely from them,

I will try testing it again on multiple platforms with Stream-Pi using system font, and report back asap
